library/Freckle/App/Bugsnag.hs

```diff
@@ -0,0 +1,164 @@
+setAppVersion AppVersion {..} = updateEvent $ \event ->
+  event { event_app = Just $ defaultApp { app_version = Just avName } }
+
+class HasBugsnagSettings env where
+  bugsnagSettingsL :: Lens' env Settings
+
+instance HasBugsnagSettings Settings where
+  bugsnagSettingsL = id
+
+instance HasBugsnagSettings site =>  HasBugsnagSettings (HandlerData child site) where
+  bugsnagSettingsL = envL . siteL . bugsnagSettingsL
+
+-- | Notify Bugsnag of an exception
+--
+-- The notification is made asynchronously via a simple @'forkIO'@. This is
+-- best-effort and we don't care to keep track of the spawned threads.
+--
+notifyBugsnag
+  :: ( MonadIO m
+     , MonadReader env m
+     , HasBugsnagSettings env
+     , Exception.Exception e
+     )
+  => e
+  -> m ()
+notifyBugsnag = notifyBugsnagWith mempty
+
+-- | 'notifyBugsnag' with a 'BeforeNotify'
+notifyBugsnagWith
+  :: ( MonadIO m
+     , MonadReader env m
+     , HasBugsnagSettings env
+     , Exception.Exception e
+     )
+  => BeforeNotify
+  -> e
+  -> m ()
+notifyBugsnagWith f ex = do
+  settings <- view bugsnagSettingsL
+  void $ liftIO $ forkIO $ Bugsnag.notifyBugsnagWith f settings ex
+
+asSqlError :: SqlError -> BeforeNotify
+asSqlError err@SqlError {..} = toSqlGrouping <> toSqlException
+ where
+  toSqlGrouping = maybe mempty setGroupingHash (sqlErrorGroupingHash err)
+  toSqlException = updateExceptions $ \ex -> ex
+    { exception_errorClass = decodeUtf8 $ "SqlError-" <> sqlState
+    , exception_message =
+      Just
+      $ decodeUtf8
+      $ sqlErrorMsg
+      <> ": "
+      <> sqlErrorDetail
+      <> " ("
+      <> sqlErrorHint
+      <> ")"
+    }
+
+sqlErrorGroupingHash :: SqlError -> Maybe Text
+sqlErrorGroupingHash err = do
+  violation <- constraintViolation err
+  decodeUtf8 <$> case violation of
+    ForeignKeyViolation table constraint -> pure $ table <> "." <> constraint
+    UniqueViolation constraint -> pure constraint
+    _ -> Nothing
+
+asHttpException :: HttpException -> BeforeNotify
+asHttpException (HttpExceptionRequest req content) =
+  setGroupingHash (decodeUtf8 $ host req) <> update
+ where
+  update = updateExceptions $ \ex -> ex
+    { exception_errorClass = "HttpExceptionRequest"
+    , exception_message =
+      Just
+      . decodeUtf8
+      $ method req
+      <> " request to "
+      <> host req
+      <> " failed: "
+      <> BS8.pack (show content)
+    }
+asHttpException (InvalidUrlException url msg) = updateExceptions $ \ex -> ex
+  { exception_errorClass = "InvalidUrlException"
+  , exception_message = Just $ pack $ url <> " is invalid: " <> msg
+  }
+
+-- | Set StackFrame's InProject to @'False'@ for Error Helper modules
+--
+-- We want exceptions grouped by the the first stack-frame that is /not/ them.
+-- Marking them as not in-project does this, with little downside.
+--
+maskErrorHelpers :: BeforeNotify
+maskErrorHelpers = setStackFramesInProjectByFile (`notElem` errorHelpers)
+  where errorHelpers = ["library/FrontRow/Core/Exceptions.hs"]
```

Comment:
This is annoying, and I'm open to the practicality argument, but `freckle-app` really shouldn't know about this list of error helpers from our actual domain-specific app.

Reply:
Yeah, totally missed that little line.

Reply:
One option is to have it check for `Exceptions` modules in general.

Reply:
For now I'm just going to codify this convention. If we have a need then we can make it more configurable.